### DISCUSSION TITLE
b-img-lazy can not load async backend called binary image data

### DISCUSSION THREAD

**Discussion by srknkcn** [bootstrap-vue/bootstrap-vue#7171] [Q&A]
URL based source is working. But when we changed with async loading from S3 bucket gives promise errors

    <b-overlay :show="spinnerShow">
        <div>
            <b-card style="height: 400px;">
                <b-row v-for="photorow in photoRowCount">
                    <b-col><b-img-lazy v-bind="mainProps" :src="getImageUrl(80)" alt="Image 1"></b-img-lazy></b-col>
                    <b-col><b-img-lazy v-bind="mainProps" :src="getImageUrl(12)" alt="Image 2"></b-img-lazy></b-col>
                    <b-col><b-img-lazy v-bind="mainProps" :src="getImageUrl(12)" alt="Image 3"></b-img-lazy></b-col>
                    <b-col><b-img-lazy v-bind="mainProps" :src="getImageUrl(13)" alt="Image 4"></b-img-lazy></b-col>
                    <b-col><b-img-lazy v-bind="mainProps" :src="getImageUrl(14)" alt="Image 5"></b-img-lazy></b-col>
                </b-row>
            </b-card>
        </div>
    </b-overlay>

    methods: {
        async getImageUrl(imageId) {
            let result = this.photoList[imageId];
            let returnvalue = await mezFotoService.GetMezFotografByName(result);
            return returnvalue;
        }
    }

and in the service section

    GetMezFotografByName = async (name) => {        
        let result = {};
        const url = `${baseUrl}/api/MezFotograf/GetMezFotografByName?name=`+name;
        await axios.get(url)
        .then((response) => {
            result = response.data.responseData;
        }).catch((error) => {
            console.log(error);
        });
        return result;
    }

All photos are requested and come to browser in the developer tools network screen.

**Comment by Hiws** [comment]
You shouldn't call async methods in the template.
